internal/cache/file/downloader/job.go

```diff
@@ -226,196 +225,186 @@ func (job *Job) notifySubscribers() {
 	}
 }
 
-// failWhileDownloading changes the status of job to failed and notifies
-// subscribers about the download error.
+// updateStatusAndNotifySubscribers is a helper function which updates the
+// status's name and err to given name and error and notifies subscribers.
 //
-// Acquires and releases LOCK(job.mu)
-func (job *Job) failWhileDownloading(downloadErr error) {
-	logger.Errorf("Job:%p (%s:/%s) failed with: %v", job, job.bucket.Name(), job.object.Name, downloadErr)
+// Acquires and releases Lock(job.mu)
+func (job *Job) updateStatusAndNotifySubscribers(statusName jobStatusName, statusErr error) {
+	if statusName == Failed {
+		logger.Errorf("Job:%p (%s:/%s) Failed with error: %v", job, job.bucket.Name(), job.object.Name, statusErr)
+	} else {
+		logger.Tracef("Job:%p (%s:/%s) status changed to %v with error: %v", job, job.bucket.Name(), job.object.Name, statusName, statusErr)
+	}
 	job.mu.Lock()
-	job.status.Err = downloadErr
-	job.status.Name = Failed
+	job.status.Err = statusErr
+	job.status.Name = statusName
 	job.notifySubscribers()
 	job.mu.Unlock()
 }
 
-// updateFileInfoCache updates the file info cache with latest offset downloaded
-// by job. Returns error in case of failure.
-//
-// Not concurrency safe and requires LOCK(job.mu)
-func (job *Job) updateFileInfoCache() (err error) {
+// updateStatusOffset updates the offset in job's status and in file info cache
+// with the given offset. If the the update is successful, this function also
+// notify the subscribers.
+func (job *Job) updateStatusOffset(downloadedOffset int64) (err error) {
 	fileInfoKey := data.FileInfoKey{
 		BucketName: job.bucket.Name(),
 		ObjectName: job.object.Name,
 	}
 	fileInfoKeyName, err := fileInfoKey.Key()
 	if err != nil {
-		err = fmt.Errorf("updateFileInfoCache: error while creating fileInfoKeyName for bucket %s and object %s %w",
+		err = fmt.Errorf("updateStatusOffset: error while creating fileInfoKeyName for bucket %s and object %s %w",
 			fileInfoKey.BucketName, fileInfoKey.ObjectName, err)
 		return
 	}
 
 	updatedFileInfo := data.FileInfo{
 		Key: fileInfoKey, ObjectGeneration: job.object.Generation,
-		FileSize: job.object.Size, Offset: uint64(job.status.Offset),
+		FileSize: job.object.Size, Offset: uint64(downloadedOffset),
 	}
-
-	logger.Tracef("Job:%p (%s:/%s) downloaded till %v offset.", job, job.bucket.Name(), job.object.Name, job.status.Offset)
+	job.mu.Lock()
+	defer job.mu.Unlock()
 	err = job.fileInfoCache.UpdateWithoutChangingOrder(fileInfoKeyName, updatedFileInfo)
-	if err != nil {
-		err = fmt.Errorf("updateFileInfoCache: error while inserting into fileInfoCache %s: %w", updatedFileInfo.Key, err)
-		return
+	if err == nil {
```

Comment:
lets use the same format everywhere
if err != nil {
  // do something
return
}

continue with other logic